eclipse-sirius/sirius-web
# [2759] Reactivate the SelectionDialog

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

- Import the Studio 
- [Studio.zip](https://github.com/user-attachments/files/16162153/Studio.zip)
- This studio is based on default one but with two additional tools:
  - One in the diagram palette which renamed the selected element
  - One on the Entity1 node that also open a selection dialog to rename the selected element
- Create a new project
- Create a new model based on the studio MM
- Create a new diagram 
- Create some element by using the node creation tools
- Use the selectionTool in the diagram contextual palette:
  - The selection dialog should be opened with all the model elements selectable
  - Select an element and perform finished
  - The element should be renamed
- Use the selection tool on an Entity1
- The selection dialog should only contain the selected element
- Select it and perform finish
- The element should be renamed
